checker/build.gradle

```diff
@@ -355,6 +355,44 @@ task demosTests(dependsOn: assembleForJavac, group: 'Verification') {
     }
 }
 
+task templateTests(dependsOn: assembleForJavac, group: 'Verification') {
+    description = 'Test that the templatefora-checker are working as expected.'
+
+    def injected = project.objects.newInstance(InjectedExecOps)
+
+    doLast {
+        File templateforCheckerDir = new File(projectDir, '../../templatefora-checker');
+        if (!templateforCheckerDir.exists()) {
+            injected.execOps.exec {
+                workingDir file(templateforCheckerDir.toString() + '/../')
+                executable 'git'
+                args = [
+                    'clone',
+                    '--depth',
+                    '1',
+                    'https://github.com/eisop/templatefora-checker.git'
+                ]
+            }
+        } else {
+            injected.execOps.exec {
+                workingDir templateforCheckerDir
+                executable 'git'
+                args = [
+                    'pull',
+                    'https://github.com/eisop/templatefora-checker.git'
```

Comment:
Regarding this [comment](https://github.com/eisop/checker-framework/pull/1048#issuecomment-2572127375):
In CI we are cloning something from the PR authors org/branch and then here we're trying to pull in the eisop version. I don't think that is correct.

We should do what we do in the `createCloneTask` in `build.gradle`: simply pull if the directory already exists.
Can you open a separate PR that reduces the duplication between the demos and template tasks by reusing the `createCloneTask` (or extracting some of its logic) and then using that here.